rusk/src/lib/http/chain.rs

```diff
@@ -253,4 +262,52 @@ impl RuskNode {
 
         Ok(ResponseData::new(serde_json::to_value(stats)?))
     }
+
+    async fn simulate_tx(&self, tx: &[u8]) -> anyhow::Result<ResponseData> {
+        let tx = ProtocolTransaction::from_slice(tx)
+            .map_err(|e| anyhow::anyhow!("Invalid transaction: {e:?}"))?;
+        let (config, mut session) = {
+            let vm_handler = self.inner().vm_handler();
+            let vm_handler = vm_handler.read().await;
+            if tx.gas_limit() > vm_handler.get_block_gas_limit() {
+                return Err(anyhow::anyhow!("Gas limit is too high."));
+            }
+            let tip = load_tip(&self.db())
+                .await
+                .map_err(|e| anyhow::anyhow!("Failed to load the tip: {e}"))?
+                .ok_or_else(|| anyhow::anyhow!("Could not find the tip"))?;
+            let height = tip.header.height;
+            let config = vm_handler.vm_config.to_execution_config(height);
+            let session = vm_handler
+                .new_block_session(height, vm_handler.tip.read().current)
+                .map_err(|e| {
+                    anyhow::anyhow!("Failed to initialize a session: {e}")
+                })?;
+            (config, session)
+        };
+        let receipt = execute(&mut session, &tx, &config)
+            .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+        let resp = match receipt.data {
+            Ok(data) => json!({
+                "gas-spent": receipt.gas_spent,
+                "data": data,
```

Comment:
execution of a transaction always return empty byte array, leaving this field as unnecessary